action.yml

```diff
@@ -175,13 +175,32 @@ runs:
             echo "CODECOV_TOKEN=$CODECOV_TOKEN" >> $GITHUB_ENV
           fi
         fi
+
+    - name: Override branch for forks
+      shell: bash
+      run: |
+        if [ -z "$CC_BRANCH" ] && [ -z "$CODECOV_TOKEN" ] && [ "${GITHUB_EVENT_PULL_REQUEST_REPO_FULL_NAME}" != "$GITHUB_REPOSITORY" ];
+        then
+          echo -e "\033[0;32m==>\033[0m Fork detected, tokenless uploading used"
+          TOKENLESS="$GITHUB_EVENT_PULL_REQUEST_HEAD_LABEL"
+          CC_BRANCH="$GITHUB_EVENT_PULL_REQUEST_HEAD_LABEL"
+          echo "TOKENLESS=$TOKENLESS" >> $GITHUB_ENV
+        fi
+
+        echo "CC_BRANCH=$CC_BRANCH" >> $GITHUB_ENV
+      env:
+        CC_BRANCH: ${{ inputs.override_branch }}
+        GITHUB_EVENT_PULL_REQUEST_HEAD_LABEL: ${{ github.event.pull_request.head.label }}
+        GITHUB_EVENT_PULL_REQUEST_REPO_FULL_NAME: ${{ github.event.pull_request.repo.full_name }}
```

Comment:
Greetings.

I am not sure where did `full_name` come from. This event variable [does not exist in the documentation](https://docs.github.com/en/rest/using-the-rest-api/github-event-types?apiVersion=2022-11-28#event-object-common-properties), and it is also not outputting anything when I am testing it in my repositories (not forks).

And due to that fact, the following check here:
`if [ -z "$CC_BRANCH" ] && [ -z "$CODECOV_TOKEN" ] && [ "${GITHUB_EVENT_PULL_REQUEST_REPO_FULL_NAME}" != "$GITHUB_REPOSITORY"`
is always false when I am upgrading to `codecov-action` v5, which leads to some side effects with the token check.

So, unless I am missing something, I think it is a bug.

Best,
Rust

Reply:
Oh, got it @thomasrockhu-codecov 
Thank you for the explanation and resolving the issue!